marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java

```diff
@@ -76,11 +84,85 @@
     @RequestMapping(value = "/{modelName}/entityTypes", method = RequestMethod.PUT)
     @ApiImplicitParam(required = true, paramType = "body", dataType = "ModelDefinitions")
     public ResponseEntity<Void> updateModelEntityTypes(@ApiParam(hidden = true) @RequestBody JsonNode entityTypes, @PathVariable String modelName) {
+        // update the model
         newService().updateModelEntityTypes(modelName, entityTypes);
+
+        //deploy updated configs
+        deployModelConfigs();
+
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
 
+    void deployModelConfigs() {
+        ManageClient manageClient = hubClientProvider.getHubClient().getManageClient();
+        DatabaseClient finalDatabaseClient = hubClientProvider.getHubClient().getFinalClient();
+        DatabaseClient stagingDatabaseClient = hubClientProvider.getHubClient().getStagingClient();
+
+        JsonNode modelConfigNode = newService().generateModelConfig();
+
+        // Deploy search options
+        String explorerOptions = modelConfigNode.get("search-options").get("explorer").asText();
+        String defaultOptions = modelConfigNode.get("search-options").get("default").asText();
+        Map<String, DatabaseClient> clientMap = new HashMap<>();
+        clientMap.put("staging", stagingDatabaseClient);
+        clientMap.put("final", finalDatabaseClient);
+        clientMap.forEach((databaseKind, databaseClient) -> {
+            QueryOptionsManager queryOptionsManager = databaseClient.newServerConfigManager().newQueryOptionsManager();
+            queryOptionsManager.writeOptionsAs("exp-" + databaseKind + "-entity-options", Format.XML, explorerOptions);
+            queryOptionsManager.writeOptionsAs(databaseKind + "-entity-options", Format.XML, defaultOptions);
+        });
+
+        // Deploy protected-paths
+        modelConfigNode.get("protected-paths").forEach(jsonNode -> manageClient.postJson("/manage/v2/protected-paths", jsonNode.toString()));
+
+        // Deploy query-rolesets
+        try {
+            modelConfigNode.get("query-rolesets").forEach(jsonNode -> manageClient.postJson("/manage/v2/query-rolesets", jsonNode.toString()));
+        }
+        catch (HttpClientErrorException ex) {
+            QueryRolesetUtil.handleSaveException(ex);
+        }
+
+        // Deploy index configuration
+        modelConfigNode.get("index").get("database-names").forEach(dbNameNode -> {
+            // Get updated entity index configs
+            JsonNode indexNode = modelConfigNode.get("index").get("index-config");
+
+            try {
+                // Get current db configuration
+                String dbConfig = manageClient.getJson("/manage/v2/databases/" + dbNameNode.asText() + "/properties");
+                JsonNode dbNode = new ObjectMapper().readTree(dbConfig);
```

Comment:
Starting with this line, put the rest into a protected method so we can unit-test it easily without depending on ML at all. This is relatively complex logic here, and we want to make sure it behaves correctly. It's easiest to do that if we can pass in our own inputs for the existing dbConfig and the generated indexNode and then verify that they are merged together correctly. Those tests will be much easier to write and will run very quickly - I recommend putting them into a separate test class that doesn't extend anything, can call it "MergeDatabaseConfigTest".

Reply:
Added the test.